Automattic/wp-calypso
# WooCommerce: Dashboard / Setup / NUX: Persist taxes-opt-out, shipping-opt-out and i'm-finished to the server

This PR adds actions to persist some of the user's dashboard/setup/nux choices (opt out of tax setup, opt out of shipping setup and i'm finished) to the server for #13612

It also does a little bit of rearranging of views to support a setup-address-view (coming in a subsequent PR) and a setup-progress-view (also coming in a subsequent PR) for #14775 

To test:
- npm run test-client client/extensions/woocommerce/state/sites/setup-choices and make sure everything passes
- Use https://developer.wordpress.com/docs/api/console/ to POST to /sites/{siteID}/calypso-preferences/woocommerce and set finished_initial_setup, opted_out_of_shipping_setup and opted_out_of_taxes_setup to 0
- Go to http://calypso.localhost:3000/store/{site} and make sure you see the task list with the shipping and taxes tasks
- Click on the "I won't be shipping" link. Make sure the shipping task goes away
- Force a refresh. Make sure the shipping task stays away
- Click on the "I'm not charging sales tax" link. Make sure the tax task goes away
- Force a refresh. Make sure the tax task stays away
- Click on the "I'm finished" button at the bottom of the page
- Make sure the dashboard content is replaced with the "no-orders-yet" view aka the view that starts with "Congratulations! Now that your setup is complete you can spread the word and get those first orders rolling in."
